### ISSUE TITLE
Adding a godot_print_rich!() macro

### ISSUE THREAD

**Issue by x2f** [godot-rust/gdext#624]
Trying to implement a logger:

      godot_print!("[color=red]some text[/color]")

...does not print a red ouput line (the string is shown as is, including the bbcode).

Would it make sense to add a godot_print_rich!() macro?

**Comment by StatisMike** [comment]
It should be noted there, that Godot's  `print_rich()` is slow, so depending on your use case this may be detrimental:

```
   -- non_godot_println              ...      2.495μs     11.139μs
   -- bench_godot_print              ...      9.303μs     21.736μs
   -- bench_godot_print_rich         ...     18.767μs     38.422μs
   -- bench_godot_print_rich_richer    ...     21.426μs     50.193μs
```

where:
print_rich:        "[color=green][b]Hello world![/b][/color]"
print_rich_richer:    "[color=green][b]Hello world![/b][/color][color=red][i] Hohoho![/i][/color]"

**Comment by x2f** [comment]
@StatisMike Thank you, this is good context info. Just wondering about the 2 columns, are they average vs. max values?

**Comment by x2f** [comment]
@Bromeon, @StatisMike, I made a pull request in case it helps.

**Comment by StatisMike** [comment]
@x2f This is an excerpt from `itest` internal crate - the first column is minimal time, and the second is median.

Unfortunately setting up the benchmarks within `itest` seems not viable - I don't think that we can inhibit printing the lines, and they obscure the output of `itest`.

**Comment by StatisMike** [comment]
@x2f It seems you opened PR inside your own fork (`x2f/gdext`), not upstream (`godot-rust/gdext`)

**Comment by x2f** [comment]
Thanks @StatisMike, hope this new PR is ok

**Comment by StatisMike** [comment]
Can be closed now as #638 is merged 😀
